tests/nnstreamer_rate/unittest_rate.cc

```diff
@@ -0,0 +1,476 @@
+
+  rate = gst_bin_get_by_name (GST_BIN (test_data.pipeline), "rate");
+  ASSERT_TRUE (rate != NULL);
+
+  g_object_set (rate, "silent", (gboolean) FALSE, NULL);
+  g_object_get (rate, "silent", &silent, NULL);
+  EXPECT_FALSE (silent);
+
+  g_object_set (rate, "throttle", (gboolean) TRUE, NULL);
+  g_object_get (rate, "throttle", &throttle, NULL);
+  EXPECT_TRUE (throttle);
+
+  g_object_set (rate, "framerate", "15/1", NULL);
+  g_object_get (rate, "framerate", &framerate, NULL);
+  EXPECT_STREQ (framerate, "15/1");
+  g_free (framerate);
+
+  gst_object_unref (rate);
+  gst_object_unref (test_data.pipeline);
+}
+
+/**
+ * @brief Test tensor_rate set property stats (negative)
+ */
+TEST (nnstreamer_rate, set_propery_stats_n)
+{
+  TestOption option;
+  GstElement *rate;
+  guint64 in, out, dup, drop;
+
+  _set_default_option (option);
+  option.mode = TENSOR_RATE_MODE_PASSTHROUGH;
+
+  ASSERT_TRUE (_setup_pipeline (option));
+
+  rate = gst_bin_get_by_name (GST_BIN (test_data.pipeline), "rate");
+  ASSERT_TRUE (rate != NULL);
+
+  g_object_set (rate, "in", 10, NULL);
+  g_object_get (rate, "in", &in, NULL);
+  EXPECT_EQ (in, DEFAULT_IN);
+
+  g_object_set (rate, "out", 10, NULL);
+  g_object_get (rate, "out", &out, NULL);
+  EXPECT_EQ (out, DEFAULT_OUT);
+
+  g_object_set (rate, "duplicate", 10, NULL);
+  g_object_get (rate, "duplicate", &dup, NULL);
+  EXPECT_EQ (dup, DEFAULT_DUP);
+
+  g_object_set (rate, "drop", 10, NULL);
+  g_object_get (rate, "drop", &drop, NULL);
+  EXPECT_EQ (drop, DEFAULT_DROP);
+
+  gst_object_unref (rate);
+  gst_object_unref (test_data.pipeline);
+}
+
+/**
+ * @brief Test tensor_rate set invalide framerate (negative)
+ */
+TEST (nnstreamer_rate, set_propery_invalid_framerate_n)
+{
+  TestOption option;
+  GstElement *rate;
+  gchar *framerate;
+
+  _set_default_option (option);
+  option.mode = TENSOR_RATE_MODE_PASSTHROUGH;
+
+  ASSERT_TRUE (_setup_pipeline (option));
+
+  rate = gst_bin_get_by_name (GST_BIN (test_data.pipeline), "rate");
+  ASSERT_TRUE (rate != NULL);
+
+  g_object_set (rate, "framerate", "ASDF", NULL);
+  g_object_get (rate, "framerate", &framerate, NULL);
+  EXPECT_STREQ (framerate, DEFAULT_SOURCE_FRAMERATE);
+  g_free (framerate);
+
+  g_object_set (rate, "framerate", "10/0", NULL);
+  g_object_get (rate, "framerate", &framerate, NULL);
+  EXPECT_STREQ (framerate, DEFAULT_SOURCE_FRAMERATE);
+  g_free (framerate);
+
+  gst_object_unref (rate);
+  gst_object_unref (test_data.pipeline);
+}
+
+/**
+ * @brief Test tensor_rate with passthrough mode
+ */
+TEST (nnstreamer_rate, passthrough)
+{
+  TestOption option;
+  GstElement *rate;
+  GstStateChangeReturn ret;
+  guint64 in, out, dup, drop;
+
+  _set_default_option (option);
+  option.mode = TENSOR_RATE_MODE_PASSTHROUGH;
+
+  ASSERT_TRUE (_setup_pipeline (option));
+
+  rate = gst_bin_get_by_name (GST_BIN (test_data.pipeline), "rate");
+  ASSERT_TRUE (rate != NULL);
+
+  ret = gst_element_set_state (test_data.pipeline, GST_STATE_PLAYING);
```

Comment:
How about using `setPipelineStateSync ()`?
It will be safer to use this even in slow environments.

Reply:
Oh, yes. I will update the codes to use `unittest_utils`.